sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -0,0 +1,587 @@
+      this.jsonMapper = jsonMapper;
+      this.responseHandler = responseHandler;
+    }
+
+    @Override
+    public RelDataType getRowType(RelDataTypeFactory typeFactory)
+    {
+      return TASKS_SIGNATURE.getRelDataType(typeFactory);
+    }
+
+    @Override
+    public TableType getJdbcTableType()
+    {
+      return TableType.SYSTEM_TABLE;
+    }
+
+    @Override
+    public Enumerable<Object[]> scan(DataContext root)
+    {
+      class TasksEnumerable extends DefaultEnumerable<Object[]>
+      {
+        private final JsonParserIterator<TaskStatusPlus> it;
+
+        public TasksEnumerable(JsonParserIterator<TaskStatusPlus> tasks)
+        {
+          this.it = tasks;
+        }
+
+        @Override
+        public Iterator<Object[]> iterator()
+        {
+          throw new UnsupportedOperationException("Do not use iterator(), it cannot be closed.");
+        }
+
+        @Override
+        public Enumerator<Object[]> enumerator()
+        {
+          return new Enumerator<Object[]>()
+          {
+            @Override
+            public Object[] current()
+            {
+              TaskStatusPlus task = it.next();
+              return new Object[]{task.getId(), task.getType(),
+                                  task.getDataSource(),
+                                  task.getCreatedTime(),
+                                  task.getQueueInsertionTime(),
+                                  task.getState(),
+                                  task.getRunnerTaskState(),
+                                  task.getDuration(),
+                                  task.getLocation() != null
+                                  ? task.getLocation().getHost() + ":" + (task.getLocation().getTlsPort()
+                                                                          == -1
+                                                                          ? task.getLocation()
+                                                                                .getPort()
+                                                                          : task.getLocation().getTlsPort())
+                                  : null,
+                                  task.getErrorMsg()};
+            }
+
+            @Override
+            public boolean moveNext()
+            {
+              return it.hasNext();
+            }
+
+            @Override
+            public void reset()
+            {
+
+            }
+
+            @Override
+            public void close()
+            {
+              try {
+                it.close();
+              }
+              catch (IOException e) {
+                throw new RuntimeException(e);
+              }
+            }
+          };
+        }
+      }
+
+      return new TasksEnumerable(getTasks(druidLeaderClient, jsonMapper, responseHandler));
+    }
+
+    //Note that overlord must be up to get tasks
+    protected JsonParserIterator<TaskStatusPlus> getTasks(
```

Comment:
this can also be a static method outside of TasksTable

Reply:
yes, made private static